### PR TITLE
Fix: access_control 404 behaviour

### PR DESCRIPTION
**Pull request by alex-schneider**

### CODE REVIEW

File: config/runtime/server.go [hunk 1]
```diff
@@ -627,20 +631,48 @@ func getPortsHostsList(hosts []string, defaultPort int) (Ports, error) {
...
 	endpoints := make(endpointMap)
 
+	apiBasePaths := make(map[string]struct{})
+
 	for _, api := range srvConf.APIs {
+		basePath := serverOptions.APIBasePaths[api]
+
+		var filesBasePath, spaBasePath string
+		if serverOptions.FilesBasePath != "" {
+			filesBasePath = path.Join("/", serverOptions.FilesBasePath)
```
Comment (malud) [inline config/runtime/server.go:644]: `NewServerOptions` ensures a leading `/` for all paths (via `options.SrvBasePath`).
This extra joins seems to be obsolete here.

**Review by malud — CHANGES_REQUESTED** [review]
Update and uncomment the other test cases for `TestAccessControl_Files_SPA` the `/favicon.ico` case still fails.

Because of using the error template instead of the `fileHandler`. Do we need `isConfigured` and `matchesPatch` if the `fileHandler` is not `nil` ?

https://github.com/avenga/couper/blob/protected-resources/server/mux.go#L155

```go
if isConfigured(m.opts.ServerOptions.FilesBasePath) &&
	matchesPath(m.opts.ServerOptions.FilesBasePath, req.URL.Path) {
		return fileHandler
		// return m.opts.ServerOptions.FilesErrTpl.ServeError(errors.FilesRouteNotFound)
}
```

File: config/runtime/server.go [hunk 2]
```diff
@@ -661,3 +693,13 @@ func parseBodyLimit(limit string) (int64, error) {
 	}
 	return units.FromHumanSize(requestBodyLimit)
 }
+
+func getACL(srvConf *config.Server, api *config.API) config.AccessControl {
```
[the remark this answers is not in the document]
Comment (alex-schneider) [reply config/runtime/server.go]: Done

File: config/runtime/server.go (hunk 1 above)
Comment (alex-schneider) [reply config/runtime/server.go]: Done

**Review by malud — APPROVED** [review]

**Pull request merged**